db/migration/V007__backfill_journal_entry_reflections.sql

```diff
@@ -1,30 +1,30 @@
-CREATE PROCEDURE createReflectionsAndAssignToJournalEntries()
-BEGIN
-    loopLabel: LOOP
-        SET @countUnassignedJournalEntries = (
-            SELECT COUNT(*)
-                FROM journal_entries
-                WHERE reflection_id IS NULL
-        );
-        IF (@countUnassignedJournalEntries = 0) THEN
-            LEAVE loopLabel;
-        END IF;
-        INSERT INTO reflections (principal_id)
-            SELECT principal_id
-            FROM journal_entries
-            WHERE reflection_id IS NULL
-            ORDER BY journal_entries.id
-            LIMIT 1;
-        UPDATE journal_entries
-            SET reflection_id = LAST_INSERT_ID()
-            WHERE reflection_id IS NULL
-            ORDER BY journal_entries.id
-            LIMIT 1;
-    END LOOP loopLabel;
-END;
-
-CALL createReflectionsAndAssignToJournalEntries();
-
-DROP PROCEDURE createReflectionsAndAssignToJournalEntries;
+-- CREATE PROCEDURE createReflectionsAndAssignToJournalEntries()
```

Comment:
Is commenting out the SQL query the "update" to the migration that you talk about in this PR's description?  I wonder what happens if you comment the SQL out, and if we need to make a new migration for this SQL task.

Reply:
@pthan1:

> Is commenting out the SQL query the "update" to the migration that you talk about in this PR's description?

Yes. Any chance to the content of a migration will result in a different checksum.

> Does this essentially remove the migration and not allow it to run?

The migration has already been run on production and will never run again against that database. The only time this SQL will be run again is in development for freshly created databases. The data migration is not necessary for a fresh database, so it doesn't matter whether it's run or not.

The reason the code is commented out and not deleted is because these migrations are kept around for posterity—to show what's been run against production. This is a weird edge case, and we're approaching the solution in a hacky way because we can't use baseline migrations because those are a paid feature.

> could we re-write these statements to work with Flyway 6?

Flyway 6 is just using an older MySQL driver. All we'd have to do to make it run properly is update the driver in the Flyway image. That said, this data migration never needs to run again, so there's no point in doing that.

Reply:
This may be simplifying the issue, but will this PR be reverted if/when the Flyway bug is fixed and Flyway is upgraded to 8?

Reply:
I can't imagine any reason why we would. A migration that's been run is dead. The only reason we keep it around is to setup new dev environments, and the version of the migration in this PR is perfectly fine for that.